### PR TITLE
Add optional <focus> element to detail

### PR DESCRIPTION
**Pull request by amstone326**

### CODE REVIEW

**Comment by wpride** [comment]
:+1: from me, waiting on @ctsims to confirm?

**Comment by amstone326** [comment]
Yea, he's gonna look through it today

File: backend/src/org/commcare/suite/model/Detail.java [hunk 1]
```diff
@@ -438,4 +442,12 @@ public void populateEvaluationContextVariables(EvaluationContext ec) {
             ec.setVariable(key, XPathFuncExpr.unpack(variables.get(key).eval(ec)));
         }
     }
+
+    public boolean evaluateFocusFunction(EvaluationContext ec) {
+        if (focusFunction == null) {
+            return false;
+        }
+        Object value = XPathFuncExpr.unpack(focusFunction.eval(ec));
+        return XPathFuncExpr.toBoolean(value).booleanValue();
```
Comment (amstone326) [inline backend/src/org/commcare/suite/model/Detail.java]: @ctsims `toBoolean` will throw an `XPathTypeMismatchException` if `value` is not convertible to a boolean, so I think that should be sufficient. does that seem right?

Comment (ctsims) [inline backend/src/org/commcare/suite/model/Detail.java]: Perfect, that sounds like the right behavior.

**Comment by ctsims** [comment]
@amstone326 Looks good, pulling these now and closing out. Can we add these to part of the unit tests to ensure that they parse correctly, etc, in the future?

**Comment by amstone326** [comment]
@ctsims yup, will add that to master

**Pull request merged**